examples/Migration/CampaignReportToCsv.cs

```diff
@@ -0,0 +1,117 @@
+{
+    /// <summary>
+    /// This code example illustrates how to use Google Ads API to get metrics about a campaign and
+    /// serialize the result as a CSV file.
+    /// </summary>
+    public class CampaignReportToCsv : ExampleBase
+    {
+        private const int PAGE_SIZE = 1000;
+
+        // Optional output file path. If left null, a file `CampaignReportToCsv.csv` will be created
+        // in the executing assembly folder.
+        private const string OUTPUT_FILE_PATH = null;
+
+        /// <summary>
+        /// Returns a description about the code example.
+        /// </summary>
+        public override string Description =>
+            "This code example illustrates how to get metrics about a campaign and serialize the " +
+            "result as a CSV file.";
+
+        /// <summary>
+        /// Runs the code example.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        public void Run(GoogleAdsClient client, long customerId,
+            string outputFilePath = OUTPUT_FILE_PATH)
+        {
+            GoogleAdsServiceClient googleAdsServiceClient =
+                client.GetService(Services.V8.GoogleAdsService);
+
+            // Create a query that retrieves campaigns.
+            string query = @"
+                SELECT
+                    campaign.id,
+                    campaign.name,
+                    segments.date,
+                    metrics.impressions,
+                    metrics.clicks,
+                    metrics.cost_micros 
+                FROM campaign 
+                WHERE segments.date DURING LAST_30_DAYS 
+                    AND campaign.status = 'ENABLED' 
+                ORDER BY segments.date DESC";
+
+            // Issues a search request with a specified page size.
+            SearchGoogleAdsRequest searchGoogleAdsRequest = new SearchGoogleAdsRequest
+            {
+                CustomerId = customerId.ToString(),
+                Query = query,
+                PageSize = PAGE_SIZE
+            };
+            PagedEnumerable<SearchGoogleAdsResponse, GoogleAdsRow> pagedResponse =
+                googleAdsServiceClient.Search(searchGoogleAdsRequest);
+
+            if (pagedResponse.Count() == 0)
+            {
+                Console.WriteLine("No results found!");
+                return;
+            }
+
+            StringBuilder csvRows = new StringBuilder();
+
+            // Set the header for the CSV file.
+            csvRows.AppendLine("Campaign.Id,Campaign.Name,Segments.Date,Metrics.Impressions," +
+                "Metrics.Clicks,Metrics.CostMicros");
+
+            // Iterate over all returned rows in all pages and extract the information.
+            foreach (GoogleAdsRow googleAdsRow in pagedResponse)
+            {
+                csvRows.AppendLine(string.Format("{0},{1},{2},{3},{4},{5}",
```

Comment:
Instead of doing string appends, can we use a library instead?

There are couple of options that I can think of.

1) There's a CSV class I wrote in Google.Ads.Common. Escapes quotes, commas etc. properly.
2) Use a library like https://joshclose.github.io/CsvHelper/. The License seems fine (Apache2, MSPL).

Reply:
Neither of these seem to be immediately appropriate choices. In Google.Api.Ads.Common.Util.CsvFile, the Headers and Records properties have no setter, and the ReadFromString method would still require this example to assemble a properly formatted CSV file as a string before just parsing it into arrays again.

I did consider using CsvHelper originally, but it would have required defining a class with properties for each field from the query, then constructing an instance for each line in the file. This seemed more elaborate than intended for the example.

Please correct me if I'm mistaken.

Reply:
Thanks for that; updated.